src/adapters/inverter/contractAdapter.ts

```diff
@@ -0,0 +1,37 @@
+import { ethers, Contract, providers } from 'ethers';
+import { logger } from '../../utils/logger';
+
+const abi = [
+  {
+    type: 'function',
+    name: 'getStaticPriceForBuying',
+    inputs: [],
+    outputs: [
+      {
+        name: '',
+        type: 'uint256',
+        internalType: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+  },
+];
+
+export class ContractAdapter {
+  private contract: Contract;
+
+  constructor(provider: providers.Provider, contractAddress: string) {
+    this.contract = new ethers.Contract(contractAddress, abi, provider);
+  }
+
+  public async getTokenPrice(): Promise<string> {
```

Comment:
Let's move it to the inverterAdapter.ts
```suggestion
  public async getTokenPrice(<whatever name>: string): Promise<string> {
```